flang/unittests/Optimizer/Builder/ComplexHelperTest.cpp

```diff
@@ -0,0 +1,64 @@
+//===- ComplexExprHelperTest.cpp -- ComplexExprHelper unit tests ----------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "flang/Optimizer/Builder/Complex.h"
+#include "gtest/gtest.h"
+#include "flang/Optimizer/Builder/FIRBuilder.h"
+#include "flang/Optimizer/Support/InitFIR.h"
+#include "flang/Optimizer/Support/KindMapping.h"
+
+struct ComplexHelperTest : public testing::Test {
+public:
+  void SetUp() override {
+    fir::KindMapping kindMap(&context,
+        "i10:80,l3:24,a1:8,r54:Double,c20:X86_FP80,r11:PPC_FP128,"
+        "r12:FP128,r13:X86_FP80,r14:Double,r15:Float,r16:Half,r23:BFloat");
+    mlir::OpBuilder builder(&context);
+    auto loc = builder.getUnknownLoc();
+
+    // Set up a Module with a dummy function operation inside.
+    // Set the insertion point in the function entry block.
+    mlir::ModuleOp mod = builder.create<mlir::ModuleOp>(loc);
+    mlir::FuncOp func = mlir::FuncOp::create(
+        loc, "func1", builder.getFunctionType(llvm::None, llvm::None));
+    auto *entryBlock = func.addEntryBlock();
+    mod.push_back(mod);
+    builder.setInsertionPointToStart(entryBlock);
+
+    fir::support::loadDialects(context);
+    firBuilder = std::make_unique<fir::FirOpBuilder>(mod, kindMap);
+  }
+
+  fir::FirOpBuilder &getBuilder() { return *firBuilder; }
+
+  mlir::MLIRContext context;
+  std::unique_ptr<fir::FirOpBuilder> firBuilder;
+};
+
+TEST_F(ComplexHelperTest, mainFunctions) {
```

Comment:
So this test will simply verify that `ComplexExprHelper` generates correct types? Could you encode this in the test name? `mainFunctions` is rather generic. Perhaps: `verifyTypes`?

Reply:
Changed the name to `verifyTypes`. At the moment, I think, we can only verify these types. Additional testing would require IR generation.

Reply:
IIUC, you can safely move this to `SetUp`:
```
  auto builder = getBuilder();
  auto loc = builder.getUnknownLoc();
  auto helper = fir::factory::Complex(builder, loc);
  auto ctx = builder.getContext();
```
It's fairly generic set-up that might be required by other tests. This way, the test itself is kept to the required minimum.

This is a nit.

Reply:
I have moved this code and some more to the `Setup` function.

Reply:
Now only the `fir::factory::Complex` `helper` is required (`firBuilder` is not) inside the test function.